src/client.ts

```diff
@@ -330,95 +309,62 @@ class OptimizelyReactSDKClient implements ReactSDKClient {
     });
   }
 
-  public getUserContext(): optimizely.OptimizelyUserContext | null {
+  public getCurrentUserContext(): optimizely.OptimizelyUserContext | null {
     if (!this._client) {
-      logger.warn(
-        'Unable to get user context because Optimizely client failed to initialize.'
-      );
+      logger.warn('Unable to get user context. Optimizely client not initialized.');
       return null;
     }
 
     if (!this.userContext) {
-      logger.warn(
-        'Unable to get user context because user was not set.'
-      );
+      logger.warn('Unable to get user context. User context not set.');
       return null;
     }
 
     return this.userContext;
   }
 
-  public getUserContextInstance(userInfo: UserInfo): optimizely.OptimizelyUserContext | null {
```

Comment:
same question about removing public method for backward compatibility?

Reply:
It's also a good callout.  This `getUserContextInstance` was trying to do too much in one call. I've separated out the responsibilities into `getCurrentUserContext` (or just `getUserContext` if you prefer), `setCurrentUserContext`, and `makeUserContextInstance` so that there's clarity on what each does.  

This major version seemed like a good time to update for clarity with less concern about backwards compatibility.  

You are reminding me, though, that I need to make the matching change suggestions in @trishahanlon 's  React docs.